pilot/pkg/model/telemetry_test.go

```diff
@@ -698,7 +698,9 @@ func TestTelemetryFilters(t *testing.T) {
 			networking.ListenerClassSidecarOutbound,
 			networking.ListenerProtocolHTTP,
 			nil,
-			map[string]string{},
+			map[string]string{
```

Comment:
Not sure this is expected. If its disabled at the root level why are we enabling alter?

Reply:
you can re-enabled with following configuration:
```yaml
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: disable-metrics
  namespace: istio-system
spec:
  metrics:
    - overrides:
        - disabled: true
          match:
            metric: ALL_METRICS
            mode: CLIENT_AND_SERVER
      providers:
        - name: prometheus
---
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: empty-prometheus
  namespace: default
spec:
  metrics:
    - providers:
        - name: prometheus
      # reenable all metrics won't work without the following override
      overrides:
        - match:
            metric: ALL_METRICS
```

@abasitt has a real use case about this.

Reply:
But thats not what the test has ?

```
	emptyPrometheus := &tpb.Telemetry{
		Metrics: []*tpb.Metrics{
			{
				Providers: []*tpb.ProviderRef{{Name: "prometheus"}},
			},
		},
	}
```

Reply:
I think we should be able to enable prometheus with following for mesh-level/namespace-level/workload-level?
```yaml
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: mesh-default
spec:
  metrics:
  - providers:
    - name: prometheus
```

Reply:
Can you explain why you think that? I don't see anything in the API that suggests that is the correct behavior.

Reply:
I can enable prometheus(even it's disabled in mesh-wide) without specific `overrides`.

Reply:
Yeah but you are not _overriding_. You are just enabling. See https://istio.io/latest/docs/tasks/observability/telemetry/#scope-inheritance-and-overrides - "Any fields specified in the namespace configuration will completely override the field from the parent configuration"

Reply:
Correct if I'm wrong, if prometheus is disabled in mesh-wide, I should be able to reenabled with following configuration as `empty-prometheus.default` will completely override `disable-metrics.istio-system`?
```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: disable-metrics
  namespace: istio-system
spec:
  metrics:
    - overrides:
        - disabled: true
          match:
            metric: ALL_METRICS
            mode: CLIENT_AND_SERVER
      providers:
        - name: prometheus
---
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: empty-prometheus
  namespace: default
spec:
  metrics:
    - providers:
        - name: prometheus
```

Reply:
```
    - overrides:
        - disabled: false
          match:
            metric: ALL_METRICS
            mode: CLIENT_AND_SERVER
      providers:
        - name: prometheus
```

if you explicitly put ^ sure. But in `empty-prometheus` you are not overriding anything since `overrides` is not set

Reply:
you're right, let me close this and resubmit another one for the test.